system/datatypes/peer_reviewed_article.php

```diff
@@ -1442,8 +1442,10 @@ protected function createApprovesDomElement() {
 	protected function processCommentText($commentText){
 		$commentHtml = htmlspecialchars($commentText);
 		$tryToParseLinks = preg_replace_callback(self::URL_REGEX, function($matches) { // replace urls with links
-			$url = $matches[0];
-			return '<a href="' . $url . '" target="_blank">' . $url . '</a>';
+			$url = $text = $matches[0];
+			if (count($matches) < 3 || !$matches[2])
+				$url = 'https://' . $url;
```

Comment:
I considered that, but that doesn't really help, since it uses http only if hypha runs on http. When hypha runs on https, the effect would still be the same and the link could still break. I also considered to always use http (and rely on the site to redirect to https), but with https being so common nowadays, I thought it would be better to just default to https always (which could be considered *slightly* safer). If you really need http because a site is http only, then you can always specify it explicitly.

Does that sound ok to you?

Reply:
>  If you really need http because a site is http only, then you can always specify it explicitly.

Thinking out loud I can imagine that some less digital savvy users will not understand to use a `http://` prefix and then a the redirect option would be less save, but cause less user difficulties.

But I trust your judgement and am fine either way. Enjoy your weekend!

Reply:
Let's stick to https, we can always revisit if we hear about problems in practice.